### DISCUSSION TITLE
3-8. 나의 이력서 v0.1 만들어보기

### DISCUSSION THREAD

**Discussion by imakerjun** [ssac-dev/yozm#47] [3장 이런 회사에 가고 싶어]
# 1. 나의 이력서 v0.1
어떤 일이든 한번에 큰 단위의 작업을 하려고하면 막연하고 어렵습니다. 이럴때 문제를 똑똑하게 잘 해결하는 사람일수록 '동작 가능한 가장 작은 버전'을 먼저 만들어보는걸로 시작하는데요. 이력서 또한 마찬가지입니다. 나의 이력서 v0.1을 만들어 봅시다.

# 2. 이력서 v0.1 만드는 방법
### 1) JD에서 핵심 키워드 뽑기
JD에서 핵심적인 역량 키워드를 먼저 뽑아볼게요. 이력서는 결국 독자가 회사 관계자 이므로, 독자가 원하는 요구사항과 질문이 있는 JD를 분석하는 과정이 필수입니다. 이때 JD를 분석하는 과정이 어렵거나 막연하다면 ChatGPT와 같은 도구를 활용해보는 것도 좋습니다.

### 2) JD의 주요 핵심 키워드와 나의 경험을 연결하기
JD의 주요 핵심 키워드와 STAR 기법으로 작성한 나의 경험을 연결해서 작성해볼게요. 이때 처음부터 모든 키워드를 다 연결하여 작성하려 하기 보다는 1개의 키워드에 대한 경험을 먼저 작성해본다는 관점으로 먼저 진행해보세요.

### 3) 소프트스킬, 학습 능력, 기술 스택 등 이력서에 포함이 되는 주요 요소 1가지 작성해보기
소프트스킬, 학습 능력, 기술 스택 3가지 중 한 가지를 선택하여 이력서에 추가로 작성해볼게요.

# 4. 실습
이제 본격적으로 이력서 v0.1를 마크다운으로 간단히 작성하여 discussion에 남겨볼게요 😄

**Comment by DragonTiger92** [comment]
회고 3조 이력서 작성 코멘트

**Reply by DragonTiger92** [reply]
# 김용범 이력서 v0.1

## 사전 정보

- 지원하고 싶은 회사 이름: 오지스(Ozys)

- 하는 일: 블록체인 기반 서비스(DApp) 개발

- 프론트엔드 직무 주니어 개발자 모집 공고 링크: https://ozys.io/career/development/2

- 첨언: 어떻게든 과제 하고 싶어서 30분 내로라도 쓸 수 있는 만큼만 쓰는 거라 여러 부분에서 미흡합니다. 참고 부탁드립니다.

## 과제 진행

### 1. JD에서 핵심 키워드 뽑기

- 프론트엔드의 안정성
- SPA 웹 애플리케이션 경험(React)
-  타 서비스 SDK나 라이브러리, API 등을 연동한 개발 경험
- 스타일 시트 렌더링 최적화

### 2. JD의 주요 핵심 키워드와 나의 경험을 연결하기

 '프론트엔드의 안정성' 키워드 관련해서 경험을 추출하고 얘기할 수 있을 것 같은데, 어떤 게 프론트엔드의 안정성 문제인지... 확실하게 떠오르는 게 없어서 작성을 보류하였습니다. 프론트엔드의 안정성에 문제가 생기는 경우가 어떤 경우일지에 대한 얘기를 들으면 그에 대한 경험을 떠올릴 수 있을 것 같습니다.
 만약 이 부분에 대해서 어떤 경위로라도 다루신다면, 그리고 과정 중 떠오른 경험에 대해 다룬다면 그 떠오른 경험을 바탕으로 STAR 기법을 이용하여 이력서 작성을 위한 경험 정리라도 해보겠습니다.

### 3. 소프트스킬, 학습 능력, 기술 스택 등 이력서에 포함이 되는 주요 요소 1가지 작성해보기

보유한 기술 스택: Git, React, ES8, SCSS
- JD에 기록된 요구 기술 스택 중 선택

**Reply by DragonTiger92** [reply]
저는 너무 대충 써서 피드백을 받을 게 없는데, 지금 피드백을 받기 위해 자료를 보강하기에는 파이널 프로젝트 때문에 시간이 부족할 것 같습니다. 나중에 다시 이력서 작성하면서 얘기해주신 것들 잘 반영해보겠습니다.
 그동안 회고 너무 재밌게 했습니다. 건강하시고, 좋은 기회로 다시 찾아뵐 수 있었으면 좋겠습니다. 오늘도 고생 많으셨습니다!

**Comment by SWLee2973** [comment]
회고 8조 코멘트

**Reply by SWLee2973** [reply]
(주)위펀 - Frontend Engineer(React)
https://www.rallit.com/positions/1602/%EC%A3%BC-%EC%9C%84%ED%8E%80-frontend-engineer-react

- 자격요건
>3년 이상의 웹 프론트엔드 개발 경험 또는 그에 준하는 역량을 보유하신 분
>최신 JavaScript(ES6 이상), HTML5, CSS3를 능숙하게 사용하시는 분
>1년 이상의 React 실무 경험
>TypeScript 를 사용한 실무 경험
>Git 을 통한 팀 단위 프로젝트 경험
>Semantic Markup, 웹 표준, 웹 접근성, 크로스 브라우징, 반응형 웹 개발에 능숙하신 분
>브라우저 렌더링, 네트워크 통신 과정의 기본적인 이해가 있으신 분

- 핵심 키워드로 선택한 항목
>3년 이상의 웹 프론트엔드 개발 경험 또는 그에 준하는 역량을 보유하신 분

```
Java/Spring framework 및 linux 서버 기반 솔루션의 기능 고도화 개발 및 서버 유지보수 업무를 4년 간 맡아왔습니다.
업무 중 기억나는 경험 중 하나는 00 부대의 기록 관리 시스템 개발 유지보수 담당자로 발령 받았을 때였습니다.

고객 특성 상 외부와의 통신이 어려워 검색 등 외부의 지원을 통한 개발이 불가능한 상황이었고,
이런 상황에서 기능 개선 및 오류 수정을 지속적으로 요청 받아 개발 및 개선을 진행했습니다.

서버 유지보수 업무 특성 상 WEB/WAS, 개발 코드-Spring/jsp는 물론 데이터베이스 쿼리 구조까지 알아야 했던 환경에서
검색을 통한 해결이 불가능했지만 기존 구현된 코드의 분석 및 활용을 통해 업무를 수행했습니다.

이 때 웹에서의 데이터 처리 속도가 느린 현상이 발견되었고, 이로 인해 데이터가 유실되는 상황이 벌어졌습니다.
이런 문제를 해결하기 위해 서버 로그와 코드 분석을 진행했고, 데이터베이스 쿼리의 join 구조가 문제임을 알아냈을 때가 기억에 남습니다.

join 구조의 변경을 통해 데이터 처리 속도가 서버 타임 아웃에 걸리는 시간(30초)에서 약 4~5초로 6배 이상 단축되어 데이터가 유실되는 문제가 해결되었습니다.

해결 못한 키워드를 기억해 퇴근 후 검색 및 공부를 통해 해결하던 상황에서 직접 도움 없이 문제가 해결되는 경험을 맛보게 되니 
그동안 역량이 차오르고 있었음을 깨달았습니다.

시작할 때에는 정말 막막한 환경에 놓였다고 걱정했지만 위의 경험을 포함한 2년 여의 경험은 제 4년 간의 경력 중 가장 개발 역량을 올려주었던 기간으로 기억합니다.

프론트엔드 개발 경험은 프론트엔드 스쿨에서의 경험 뿐이지만, 위의 경험이 요구하시는 역량을 보유했다고 판단되면 좋겠습니다.
```

- 느낀점
chat gpt를 통해 작성해 보면 react를 4년동안 쓴 것처럼 작성을 해주어서 일단은 직접 작성했어요.
프론트엔드 신입 공고가 거의 없고.. 1~3년 주니어 공고를 찾아서 JD를 확인해 보니 맞출 수 있는 부분이 역량에 대한 부분인 것으로 생각해서 적었습니다. 그런데 적고 보니 제가 서버 유지보수 엔지니어로 4년 2개월 경력인데 프론트엔드 개발자로서의 역량을 아예 빼고 적는 것 같아서 걱정입니다..
경력 기간 동안의 경험을 이런 식으로 작성하면 될까요?

**Comment by bellori729** [comment]
회고 7조 코멘트

**Comment by junghyunlee0922** [comment]
회고 6조 코맨트

**Comment by dldnlee** [comment]
회고 2조 코멘트

**Comment by clapsheep** [comment]
회고 5조 코멘트

**Reply by clapsheep** [reply]
# 박수양 이력서 v0.1
참고 공고 - [(주)미리디 채용공고](https://www.jobplanet.co.kr/job/search?posting_ids%5B%5D=1265949)

## 미리디 JS
주요 업무
• Typescript, React, Recoil, Emotion 기반의 내부 백오피스 웹 프론트엔드 개발
• Three.js 활용 3D 편집툴 개발
• AI 기반의 새로운 신규 서비스들을 웹으로 제공하기위한 연구 / 개발

자격 요건
• 원하는 웹 페이지와 동작을 구현할 수 있는 html, css, javascript 기초지식
• react, typescript 활용 개발 경험
• 객체 지향 프로그래밍에 대한 이해
• 자료구조, 운영체제, 컴퓨터 아키텍처, 네트워크 환경에 대한 이해
• 기본적인 git 활용
**• 빠르게 자주 바뀌는 프론트엔드의 새로운 기술 트렌드를 적극적으로 학습 / 활용할 수 있는 의지**

우대사항
"이런 분이라면 장기적으로 서로에게 더 긍정적일 것 같아요!"
• 관련학과 전공
**• 복잡한 프론트엔드 웹앱의 상태관리 경험**
• HTML canvas, WebGL, WebGPU, WebAssembly 등을 활용한 이미지 처리 프로그램 개발 경험
• scss, emotion 활용해서 css 를 작성 / 관리해본 경험
**• 번들링, 메모리 최적화를 개선하기위한 노력을 해본 경험**
• 모바일 앱 개발 경험
• AWS 등 클라우드 개발 환경 경험
• 포토샵, 일러스트, 블렌더등 그래픽 편집 도구 활용 경험
• Node.js, JAVA, Python, Rust 백엔드 개발 경험
• 프론트엔드 CI/CD 구축 경험
• Docker 활용한 개발 환경 구성 경험

## 핵심키워드
**• 빠르게 자주 바뀌는 프론트엔드의 새로운 기술 트렌드를 적극적으로 학습 / 활용할 수 있는 의지**
**• 복잡한 프론트엔드 웹앱의 상태관리 경험**
**• 번들링, 메모리 최적화를 개선하기위한 노력을 해본 경험**

## 작성한 이력서 - JD 및 STAR 기법을 참조한 ChatGPT가 만든 내용
```
"저의 프론트엔드 개발에 대한 열정은 새로운 기술 트렌드를 적극적으로 학습하고 활용하는 데에 있습니다.
최근 진행한 프로젝트에서는 많은 이미지를 처리해야 하는 서비스를 개발하는 과정에서, 데이터 캐싱의 필요성을 인지하게 되었습니다.
이에 한 번도 경험이 없던 React-query라는 상태 관리 라이브러리를 도입하고자 했습니다.

React-query의 활용을 위해 저는 공식 문서를 철저히 읽고 라이브러리의 사용법을 익혔습니다.
더불어, 실제로 캐싱이 필요한 데이터를 분석하고 적용하는 과정에서는 끊임없는 고민과 노력을 기울였습니다.
만일 어려움에 직면할 경우, 개발자 커뮤니티에서 적극적인 소통을 통해 해결책을 모색하였습니다.

결과적으로, 데이터 캐싱을 통해 이미지가 가장 많이 나오는 리스트 페이지의 렌더링 속도를 개선했으며,
추가적으로 최초 데이터를 불러오는 동안 레이아웃이 무너지는 불편함을 발견하여 사용자 경험의 만족도를 더욱 증대시킬 수 있었습니다.
이러한 경험을 통해 저는 빠르게 변화하는 프론트엔드 기술 트렌드에 대한 적극적인 의지를 보여드리고,
회사의 프로젝트에 적극적으로 기여할 수 있음을 자부합니다."
```

## 느낀점
아직 다양한 경험이 없고, 회사에서 요구하는 기술 스택을 많이 사용해보지 않아 기술적인 경험을 담은 이력서 작성이 힘들었습니다.
그래서 파이널 때, 경험해볼 수 있는 상황을 가정해서 STAR기법에 적용 가능한 상황을 만들었습니다.

시간이 부족해서 chat GPT가 만든 결과를 그대로 제출하지만,
제가 작성한다면 상황(Situation)과 과정(Task)은 면접관이 빠르게 파악할 수 있게 더욱 간결하고 명확하게 작성할 것 같고,
무엇보다 행동(Action)에 초점을 두어 제가 어떤 행동을 통해 해당 회사가 요구하는 조건에 부합한 사람일지 어필할 것 같습니다.
또한 결과(Result)는 조금 더 구체적으로 수치화해서 명확한 결과물에 대한 자료를 제공하는 방법이 좋을 것 같다 생각했습니다.
